[WIP]: IBMZ updates for HCP 4.17 (#83206)
openshift/openshift-docs: modules/hcp-ibmz-update-reg-ca.adoc

:_mod-docs-content-type: CONCEPT
[id="hcp-ibmz-update-registry-ca-private-registry_{context}"]

> = Update Registry Certificate Authority with Private Registry

Reviewer: instead of Private "Registry" use "Mirror Registry"

> = Update Registry Certificate Authority with Private Registry

Reviewer: Change the heading to "Update Registry Certificate Authority in AgentServiceConfig with Mirror Registry "

Author: Fixed

> . After you install the Multi Cluster Engine (MCE),  you must create a 'ConfigMap' in the MCE namespace with the private registry details. This ensures that you grant the hosted cluster workers the capability to retrieve images from the private registry.

Reviewer: instead of "private registry" use "mirror registry"

Author: Fixed

> . After you install the Multi Cluster Engine (MCE),  you must create a 'ConfigMap' in the MCE namespace with the private registry details. This ensures that you grant the hosted cluster workers the capability to retrieve images from the private registry.

Reviewer: After the installation of MultiCluster Engine (MCE) operator, create 'ConfigMap' with the mirror registry details under the same namespace where MCE is installed .

Author: Fixed

+
.Example ConfigMap file
[source,terminal]
----
apiVersion: v1
kind: ConfigMap
metadata:
  name: mirror-config
  namespace: multicluster-engine
  labels:
    app: assisted-service
data:
  ca-bundle.crt: |
    -----BEGIN CERTIFICATE-----
    -----END CERTIFICATE-----
  registries.conf: |

    [[registry]]
      location = "registry.stage.redhat.io"
      insecure = false
      blocked = false
      mirror-by-digest-only = true
      prefix = ""

      [[registry.mirror]]
        location = "<mirror_registry>"
        insecure = false

    [[registry]]
      location = "registry.redhat.io/multicluster-engine"
      insecure = false
      blocked = false
      mirror-by-digest-only = true
      prefix = ""

      [[registry.mirror]]
        location = "<mirror_registry>/multicluster-engine"
        insecure = false
----

> . Patch the `agentserviceconfig` resource to include the already created `ConfigMap`. If the `agentserviceconfig` resource is not present, create the `AgentServiceConfig` resource with the following content embedded into it:

Author: <img width="639" alt="image" src="https://github.com/user-attachments/assets/b82d1fc5-217b-4dcd-aa28-17f8853911c8">
Appearing as Step 2 in source

:_mod-docs-content-type: CONCEPT
[id="hcp-ibmz-update-registry-ca-private-registry_{context}"]

> = Update Registry Certificate Authority in AgentServiceConfig with Mirror Registry

Reviewer 2: ```suggestion
= Update the registry certificate authority in the AgentServiceConfig resource with the mirror registry
```
Check my revision for accuracy.

> When using a mirror registry for images, agents need to trust the registry's Certificate to securely pull images. For this, we must add the Certificate Authority of mirror registry to `AgentServiceConfig` by creating a `ConfigMap`.

Reviewer 2: ```suggestion
When you use a mirror registry for images, agents need to trust the registry's certificate to securely pull images. Add the certificate authority of mirror registry to the `AgentServiceConfig` custom resource by creating a config map.
```

> 

Reviewer 2: Add a `.Prerequisites` section to state that you must have installed the MCE Operator. When you refer to the MCE Operator, use the {mce} or {mce-short} attributes. For example:

```
.Prerequisites

* You must have installed {mce}.
```

Then, before the procedure begins, add a `.Procedure` heading per the modular documentation guidelines.

. After the installation of MultiCluster Engine (MCE) operator, create 'ConfigMap' with the mirror registry details under the same namespace where MCE is installed. This ensures that you grant the hosted cluster workers the capability to retrieve images from the mirror registry.
+
.Example ConfigMap file
[source,terminal]
----
apiVersion: v1
kind: ConfigMap
metadata:
  name: mirror-config
  namespace: multicluster-engine
  labels:
    app: assisted-service
data:
  ca-bundle.crt: |
    -----BEGIN CERTIFICATE-----
    -----END CERTIFICATE-----
  registries.conf: |

    [[registry]]
      location = "registry.stage.redhat.io"
      insecure = false
      blocked = false
      mirror-by-digest-only = true
      prefix = ""

      [[registry.mirror]]
        location = "<mirror_registry>"
        insecure = false

    [[registry]]
      location = "registry.redhat.io/multicluster-engine"
      insecure = false
      blocked = false
      mirror-by-digest-only = true
      prefix = ""

      [[registry.mirror]]

>         location = "<mirror_registry>/multicluster-engine"

Reviewer 2: If the user needs to replace any variables in this code, be sure to include callouts to explain what needs to be replaced.

        insecure = false
----

> . Patch the `agentserviceconfig` resource to include the already created `ConfigMap`. If the `agentserviceconfig` resource is not present, create the `AgentServiceConfig` resource with the following content embedded into it:

Reviewer 2: Add a blank line between lines 46 and 47.

> . Patch the `agentserviceconfig` resource to include the already created `ConfigMap`. If the `agentserviceconfig` resource is not present, create the `AgentServiceConfig` resource with the following content embedded into it:

Reviewer 2: ```suggestion
. Patch the `AgentServiceConfig` resource to include the config map that you created. If the `AgentServiceConfig` resource is not present, create the `AgentServiceConfig` resource with the following content embedded into it:
```

:_mod-docs-content-type: CONCEPT
[id="hcp-ibmz-update-registry-ca-private-registry_{context}"]
= Update the registry certificate authority in the AgentServiceConfig resource with the mirror registry

When you use a mirror registry for images, agents need to trust the registry's certificate to securely pull images. Add the certificate authority of mirror registry to the `AgentServiceConfig` custom resource by creating a `ConfigMap`.

.Prerequisites

* You must have installed {mce}.

.Procedure 

> . After the installation of MultiCluster Engine (MCE) operator, create 'ConfigMap' with the mirror registry details under the same namespace where MCE is installed. This ensures that you grant the hosted cluster workers the capability to retrieve images from the mirror registry.

Reviewer 2: ```suggestion
. In the same namespace where you installed {mce-short}, create a config map with the mirror registry details. This config map ensures that you grant the hosted cluster workers the capability to retrieve images from the mirror registry.
```

+
.Example ConfigMap file
[source,terminal]
----
apiVersion: v1
kind: ConfigMap
metadata:
  name: mirror-config
  namespace: multicluster-engine
  labels:
    app: assisted-service
data:
  ca-bundle.crt: |
    -----BEGIN CERTIFICATE-----
    -----END CERTIFICATE-----
  registries.conf: |

    [[registry]]
      location = "registry.stage.redhat.io"
      insecure = false
      blocked = false
      mirror-by-digest-only = true
      prefix = ""

      [[registry.mirror]]
        location = "<mirror_registry>"
        insecure = false

    [[registry]]
      location = "registry.redhat.io/multicluster-engine"
      insecure = false
      blocked = false
      mirror-by-digest-only = true
      prefix = ""

      [[registry.mirror]]
        location = "<mirror_registry>/multicluster-engine" // <1>
        insecure = false

> ----

Reviewer 2: Add a plus sign (+) between line 52 and line 53.

:_mod-docs-content-type: CONCEPT

> [id="hcp-ibmz-update-registry-ca-private-registry_{context}"]

Reviewer 3: ```suggestion
[id="hcp-ibmz-update-registry-ca_{context}"]
```

Reviewer 3: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#additional-resources-sections:~:text=The%20anchor%20ID%20with%20_%7Bcontext%7D%20must%20be%20lowercase%20and%20match%20the%20module%E2%80%99s%20file%20name.

= Update the registry certificate authority in the AgentServiceConfig resource with the mirror registry

> When you use a mirror registry for images, agents need to trust the registry's certificate to securely pull images. Add the certificate authority of mirror registry to the `AgentServiceConfig` custom resource by creating a `ConfigMap`.

Reviewer 3: ```suggestion
When you use a mirror registry for images, agents need to trust the registry's certificate to securely pull images. You can add the certificate authority of the mirror registry to the `AgentServiceConfig` custom resource by creating a `ConfigMap`.
```

.Prerequisites

* You must have installed {mce}.

.Procedure 

> . In the same namespace where you installed {mce-short}, create a config map with the mirror registry details. This config map ensures that you grant the hosted cluster workers the capability to retrieve images from the mirror registry.

Reviewer 3: ```suggestion
. In the same namespace where you installed {mce-short}, create a `ConfigMap` resource with the mirror registry details. This `ConfigMap` resource ensures that you grant the hosted cluster workers the capability to retrieve images from the mirror registry.
```

Reviewer 3: I _think_ in these instances you are referring to a specific `ConfigMap` resource and not the general sense of config maps.

+
.Example ConfigMap file

> [source,terminal]

Reviewer 3: ```suggestion
[source,yaml]
```

Reviewer 3: I'm pretty sure this should YAML not terminal.

----
apiVersion: v1
kind: ConfigMap
metadata:
  name: mirror-config
  namespace: multicluster-engine
  labels:
    app: assisted-service
data:
  ca-bundle.crt: |
    -----BEGIN CERTIFICATE-----
    -----END CERTIFICATE-----
  registries.conf: |

    [[registry]]
      location = "registry.stage.redhat.io"
      insecure = false
      blocked = false
      mirror-by-digest-only = true
      prefix = ""

      [[registry.mirror]]
        location = "<mirror_registry>"
        insecure = false

    [[registry]]
      location = "registry.redhat.io/multicluster-engine"
      insecure = false
      blocked = false
      mirror-by-digest-only = true
      prefix = ""

      [[registry.mirror]]

>         location = "<mirror_registry>/multicluster-engine" // <1>

Reviewer 3: ```suggestion
        location = "<mirror_registry>/multicluster-engine" # <1>
```

        insecure = false
----
+

> <1> Provide the name of the mirror registry.

Reviewer 3: ```suggestion
<1> Where: `<mirror_registry>` is the name of the mirror registry.
```

> .  Patch the `AgentServiceConfig` resource to include the config map that you created. If the `AgentServiceConfig` resource is not present, create the `AgentServiceConfig` resource with the following content embedded into it:

Reviewer 3: ```suggestion
.  Patch the `AgentServiceConfig` resource to include the `ConfigMap` resource that you created. If the `AgentServiceConfig` resource is not present, create the `AgentServiceConfig` resource with the following content embedded into it:
```

Reviewer 3: You might include an optional step here that shows how to create the `AgentServiceConfig` resource.
